datacube_stats/statistics.py

```diff
@@ -1170,5 +1229,49 @@ def _vars_to_transpose(data):
                 return ('latitude', 'longitude', 'variable', 'time'), ('variable', 'latitude', 'longitude')
 
     STATS['new_geomedian'] = NewGeomedianStatistic
+
+except ImportError:
+    pass
+
+try:
+    from hdmedians import nangeomedian
+
+    class PreciseGeoMedian(Statistic):
```

Comment:
This is a duplication of `GeoMedian` and shouldn't exist. It would be trivial to add `maxiters` as an argument to it.

Also, this is missing some fixes that were included in `GeoMedian` for handling geographic (lat/lon) data.